gito/core.py

```
@@ -66,6 +67,45 @@ def commit_in_branch(repo: Repo, commit: Commit, target_branch: str) -> bool:
+            logging.info(
+                "Checking if repo has 'main' or 'master' branchs to use as --against branch..."
+            )
+            remote_refs = repo.remotes.origin.refs
+            for branch_name in ['main', 'master']:
+                if hasattr(remote_refs, branch_name):
+                    return f'origin/{branch_name}'
+        except Exception:
+            pass
+
+        logging.error("Could not determine default branch from remote refs.")
+        raise ValueError("No default branch found in the repository.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)